### PR TITLE
Use new Harfbuzz array import/export to speed up encode and decode.

### PR DESCRIPTION
**Pull request by andyjgf**

### CODE REVIEW

File: patch_subset/sparse_bit_set.cc [hunk 1]
```diff
@@ -76,6 +76,7 @@ StatusCode SparseBitSet::Decode(string_view sparse_bit_set, hb_set_t* out) {
   uint64_t node_base_factor = leaf_node_size >> kBFNodeSizeLog2[branch_factor];
   vector<uint32_t> node_bases{0u};  // Root node.
   vector<uint32_t> next_level_node_bases;
+  vector<uint32_t> pending_codepoints;
```
Comment (garretrieger) [inline patch_subset/sparse_bit_set.cc:79]: prefer to have this be vector<hb_codepoint_t> which matches what  the set api takes.

File: patch_subset/sparse_bit_set.cc [hunk 2]
```diff
@@ -684,10 +689,10 @@ string SparseBitSet::Encode(const hb_set_t& set) {
     return "";
   }
   vector<uint32_t> codepoints;
```
Comment (garretrieger) [inline patch_subset/sparse_bit_set.cc:691]: Same here.

**Review by garretrieger — APPROVED** [review]

File: patch_subset/sparse_bit_set.cc (hunk 1 above)
Comment (andyjgf) [reply patch_subset/sparse_bit_set.cc]: Done.

File: patch_subset/sparse_bit_set.cc (hunk 2 above)
Comment (andyjgf) [reply patch_subset/sparse_bit_set.cc]: Done

**Pull request merged**